### DISCUSSION TITLE
MQTT Tunneling through HTTP proxy using client certificate to log in to proxy

### DISCUSSION THREAD

**Discussion by swinterSIE** [eclipse-paho/paho.mqtt.c#1467] [Q&A] [answered]
Hello together,

I have a question about tunneling the MQTT communication through an HTTP proxy. I am aware that paho mqtt supports proxies with username/password authentication with _proxy_set. But in my case, I want to use a client certificate plus the basic authentication to authenticate at the proxy server. Is anyone aware if this is possible or how to do it?

Thank you and kind regards,
Sven

**Comment by jumoog** [comment]
I've never tried anything like that either. But I can imagine that the proxy needs an extra configuration for this case

**Reply by swinterSIE** [reply, marked as answer]
Hi jumoog, 

thanks for the quick response. I don't know much about how the proxy handles it, but I successfully tested the proxy with a simple curl request using a client certificate. So from proxy perspective this should work. I'm just curious if there is a way to send those certificates from the paho client. Maybe in a similar way like you can do with the certificates to connect to the MQTT broker. But if you don't know about a solution, I think I have to find a different approach.
Thank you, though!

Kind regards.
